docs/services/github.md

```diff
@@ -61,12 +63,68 @@ metadata:
 ![](https://user-images.githubusercontent.com/18019529/108520497-168ce180-730e-11eb-93cb-b0b91f99bdc5.png)
 
 ```yaml
+template.app-sync-succeeded: |
+ message: |
+   Application {{.app.metadata.name}} sync succeeded at {{.app.status.operationState.finishedAt}}.
+ github:
+   repoURL: "{{.app.spec.source.repoURL}}"
+   revision: "{{.app.status.operationState.operation.sync.revision}}"
+   status:
+     state: success
+     label: "continuous-delivery/{{.app.metadata.name}}"
+     targetURL: "{{.context.argocdUrl}}/applications/{{.app.metadata.name}}?operation=true"
+   pullRequestComment:
+     content: |
+       Sync Succeeded
+
+template.app-sync-failed: |
+ message: |
+   Application {{.app.metadata.name}} sync failed at {{.app.status.operationState.finishedAt}}.
+ github:
+   repoURL: "{{.app.spec.source.repoURL}}"
+   revision: "{{.app.status.operationState.operation.sync.revision}}"
+   status:
+     state: failure
+     label: "continuous-delivery/{{.app.metadata.name}}"
+     targetURL: "{{.context.argocdUrl}}/applications/{{.app.metadata.name}}?operation=true"
+   pullRequestComment:
+     content: |
+       Sync Failed
+
+template.app-sync-running: |
+ message: |
+   Application {{.app.metadata.name}} sync started at {{.app.status.operationState.startedAt}}.
+ github:
+   repoURL: "{{.app.spec.source.repoURL}}"
+   revision: "{{.app.status.operationState.operation.sync.revision}}"
+   status:
+     state: pending
+     label: "continuous-delivery/{{.app.metadata.name}}"
+     targetURL: "{{.context.argocdUrl}}/applications/{{.app.metadata.name}}?operation=true"
+   pullRequestComment:
+     content: |
+       Sync Running
+
+template.app-sync-status-unknown: |
+ message: |
+   Application {{.app.metadata.name}} sync status is unknown as of {{.app.status.operationState.finishedAt}}.
+ github:
+   repoURL: "{{.app.spec.source.repoURL}}"
+   revision: "{{.app.status.operationState.operation.sync.revision}}"
+   status:
+     state: error
+     label: "continuous-delivery/{{.app.metadata.name}}"
+     targetURL: "{{.context.argocdUrl}}/applications/{{.app.metadata.name}}?operation=true"
+   pullRequestComment:
+     content: |
+       Sync Status Unknown
+
 template.app-deployed: |
   message: |
     Application {{.app.metadata.name}} is now running new version of deployments manifests.
   github:
     repoURLPath: "{{.app.spec.source.repoURL}}"
-    revisionPath: "{{.app.status.operationState.syncResult.revision}}"
+    revisionPath: "{{.app.status.operationState.operation.sync.revision}}"
```

Comment:
why would you take it from operation instead syncResult?

Reply:
Because it did not work otherwise with my setup, and I could not tell why ! In my `Application` manifests, `syncResult` was missing for some reasons. I double checked this time and I can now find it. What is the difference between both ? Why would this happen ?

Reply:
syncResult exist when sync already finished. It may not exist if application is just created and sync still didnt happen before